hello-data-deployment/docker-compose/base/jupyterhub/jupyterhub_config.py

```diff
@@ -0,0 +1,218 @@
+# Configuration file for JupyterHub
+import jwt
+import logging
+import os
+import requests
+from base64 import b64decode
+from cryptography.hazmat.primitives import serialization
+from jupyterhub.app import JupyterHub
+from jupyterhub.handlers import BaseHandler
+from oauthenticator.generic import GenericOAuthenticator
+from tornado import web
+from traitlets import default
+
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+c = get_config()  # noqa: F821
+
+# We rely on environment variables to configure JupyterHub so that we
+# avoid having to rebuild the JupyterHub container every time we change a
+# configuration parameter.
+
+# Spawn single-user servers as Docker containers
+c.JupyterHub.spawner_class = "dockerspawner.DockerSpawner"
```

Comment:
On Cloud deployment we use instead of DockerSpawner the `KubeSpawner` as `profile_list` see https://git.mgmtbi.ch/kubernetes/gitops/public/helm_charts/-/merge_requests/226/diffs#e5f164e73bbc0ee61aad9e18b87d28f6844d7bf1_0_54.

E.g. we use existing dockerimage either already existing such as `jupyter/datascience-notebook:latest` or `jupyter/r-notebook:x86_64-python-3.11`. or otherwise from Workspaces, where we already install all the necessary python libraries.


@wieczorslawo: What are you mapping into the dockerfile atm that we must have in the dockerimage? 

E.g. [append_csp.py](https://github.com/kanton-bern/hellodata-be/pull/83/files#diff-ac2106f0792a8ad2181a982c0ca3f73c5497c9c0dc9c80b94f76c4ff1c88ef52), wouldn't that be something that needs to be inside the sidecar of Jupyterhub, where the Sidecar would set that "authentication settings" as well as the Postgres User.

Reply:
The sidecar doesn't know anything about spawned pods/containers so it's not possible. This config needs to go directly to the notebook image or it's pod config via env var or what have you.

https://jupyter-notebook.readthedocs.io/en/5.7.1/public_server.html#embedding-the-notebook-in-another-website
https://discourse.jupyter.org/t/csp-jupyterlab4/20917/5
https://github.com/jupyter/notebook/issues/886

Reply:
> "What are you mapping into the dockerfile atm that we must have in the dockerimage"

 - the CSP config, 
 - the Python script to connect to the database with a temp user
 - the readme file on how to use the script
 That's how it's made for the docker-compose